File: C2/PDOO/Practicas/DeepSpaceRuby/lib/HangarToUI.rb
#encoding:utf-8

module Deepspace

# 2.3.17 - Translation from Java
# @author Profesor
    
class HangarToUI 
  attr_reader :maxElements, :weapons, :shieldBoosters
    
  def initialize(h) 
    @weapons=Array.new()
    @shieldBoosters=Array.new()
    @maxElements=h.maxElements
    
    for w in h.weapons do
      @weapons.push(w.getUIversion())
    end
    
    for s in h.shieldBoosters do
      @shieldBoosters.push(s.getUIversion())
    end
  end
  
  def to_s
    return "Max Elements: #{@maxElements}, Weapons: [#{@weapons.join(", ")}], Shields: [#{@shieldBoosters.join(", ")}]"
  end

end # class

end # module

File: C2/PDOO/Practicas/DeepSpaceJava/src/deepspace/HangarToUI.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

import java.util.ArrayList;

/**
 *
 * @author Profesor
 */
public class HangarToUI {
    private int maxElements;
    private ArrayList<WeaponToUI> weapons;
    private ArrayList<ShieldToUI> shieldBoosters;    
    
    HangarToUI(Hangar h) {
        weapons=new ArrayList();
        shieldBoosters=new ArrayList(); 
        
        maxElements=h.getMaxElements();
        
        for (Weapon w:h.getWeapons()) {
            weapons.add(w.getUIversion());
        }
        
        for(ShieldBooster s:h.getShieldBoosters()) {
            shieldBoosters.add(s.getUIversion());
        }
    }

    public int getMaxElements() {
        return maxElements;
    }

    public ArrayList<WeaponToUI> getWeapons() {
        return weapons;
    }

    public ArrayList<ShieldToUI> getShieldBoosters() {
        return shieldBoosters;
    }
    
    
}
